api/resources_portal/models/user.py

```diff
@@ -1,18 +1,32 @@
 import uuid
 
 from django.conf import settings
-from django.contrib.auth.models import AbstractUser
+from django.contrib.auth.models import AbstractUser, UserManager
 from django.db import models
 from django.db.models.signals import post_save
 from django.dispatch import receiver
 from rest_framework.authtoken.models import Token
 
 
+class NonDeletedObjectsManager(UserManager):
+    def get_queryset(self):
+        return super().get_queryset().filter(deleted=False)
+
+
+class DeletedObjectsManager(models.Manager):
+    def get_queryset(self):
+        return super().get_queryset().filter(deleted=True)
+
+
 class User(AbstractUser):
+    objects = NonDeletedObjectsManager()
+    deleted_objects = DeletedObjectsManager()
```

Comment:
Why are you doing it like this instead of with `safedelete` like the other models?

Reply:
safe-delete doesn't play nice with the AbstractUser parent class. I looked up the code for AbstractUser on GitHub, and tried quite a few ways of inheriting/combining AbstractUser and SafeDeleteModel, but nothing seemed to work. Anything that directly overrides the "objects" property of the model causes failure. I also tried some solutions where I combined the UserManager and SafeDeleteManager, but no bones on that either.

Reply:
Is it important enough that I should give it some more effort?